### ISSUE TITLE
problem with redeploy in kubernetes

### ISSUE THREAD

**Issue by Aleks3050** [AdaptiveConsulting/nexus-casc-plugin#2]
Hi @mattbarrett !
We try to use your plugin in kubernetes with chart https://github.com/Oteemo/charts/tree/master/charts/sonatype-nexus, nexus version sonatype/nexus3:3.24.0 in statefull mode.
First time deploy - all work fine. 
When we change config for plugin and try to deploy we see following error:
`2020-07-20 15:01:40,170+0000 ERROR [FelixStartLevel] *SYSTEM com.weareadaptive.nexus.casc.plugin.internal.NexusCascPlugin - Failed transition: NEW -> STARTED
java.lang.NullPointerException: null
        at com.weareadaptive.nexus.casc.plugin.internal.NexusCascPlugin.lambda$4(NexusCascPlugin.java:236)
        at java.util.ArrayList.forEach(ArrayList.java:1257)
        at com.weareadaptive.nexus.casc.plugin.internal.NexusCascPlugin.applyRepositoryConfig(NexusCascPlugin.java:222)
        at com.weareadaptive.nexus.casc.plugin.internal.NexusCascPlugin.doStart(NexusCascPlugin.java:121)
        at org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport.start(StateGuardLifecycleSupport.java:69)
        at org.sonatype.nexus.common.stateguard.MethodInvocationAction.run(MethodInvocationAction.java:39)
        at org.sonatype.nexus.common.stateguard.StateGuard$TransitionImpl.run(StateGuard.java:193)
        at org.sonatype.nexus.common.stateguard.TransitionsInterceptor.invoke(TransitionsInterceptor.java:56)
        at org.sonatype.nexus.extender.NexusLifecycleManager.startComponent(NexusLifecycleManager.java:199)
        at org.sonatype.nexus.extender.NexusLifecycleManager.to(NexusLifecycleManager.java:111)
        at org.sonatype.nexus.extender.NexusContextListener.moveToPhase(NexusContextListener.java:321)
        at org.sonatype.nexus.extender.NexusContextListener.frameworkEvent(NexusContextListener.java:244)
        at org.apache.felix.framework.Felix.setActiveStartLevel(Felix.java:1431)
        at org.apache.felix.framework.FrameworkStartLevelImpl.run(FrameworkStartLevelImpl.java:308)
        at java.lang.Thread.run(Thread.java:748)
2020-07-20 15:01:40,172+0000 WARN  [FelixStartLevel] *SYSTEM org.sonatype.nexus.extender.NexusLifecycleManager - Problem starting TASKS: com.weareadaptive.nexus.casc.plugin.internal.NexusCascPlugin$$EnhancerByGuice$$5fcbb89e@2f9aa84c
java.lang.NullPointerException: null
        at com.weareadaptive.nexus.casc.plugin.internal.NexusCascPlugin.lambda$4(NexusCascPlugin.java:236)
        at java.util.ArrayList.forEach(ArrayList.java:1257)
        at com.weareadaptive.nexus.casc.plugin.internal.NexusCascPlugin.applyRepositoryConfig(NexusCascPlugin.java:222)
        at com.weareadaptive.nexus.casc.plugin.internal.NexusCascPlugin.doStart(NexusCascPlugin.java:121)
        at org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport.start(StateGuardLifecycleSupport.java:69)
        at org.sonatype.nexus.common.stateguard.MethodInvocationAction.run(MethodInvocationAction.java:39)
        at org.sonatype.nexus.common.stateguard.StateGuard$TransitionImpl.run(StateGuard.java:193)
        at org.sonatype.nexus.common.stateguard.TransitionsInterceptor.invoke(TransitionsInterceptor.java:56)
        at org.sonatype.nexus.extender.NexusLifecycleManager.startComponent(NexusLifecycleManager.java:199)
        at org.sonatype.nexus.extender.NexusLifecycleManager.to(NexusLifecycleManager.java:111)
        at org.sonatype.nexus.extender.NexusContextListener.moveToPhase(NexusContextListener.java:321)
        at org.sonatype.nexus.extender.NexusContextListener.frameworkEvent(NexusContextListener.java:244)
        at org.apache.felix.framework.Felix.setActiveStartLevel(Felix.java:1431)
        at org.apache.felix.framework.FrameworkStartLevelImpl.run(FrameworkStartLevelImpl.java:308)
        at java.lang.Thread.run(Thread.java:748)`

When we use deployment mode(without pvc) all work fine, redeploy corrected.
Can you help us with this issue?

**Comment by bdellegrazie** [comment]
Hello @Aleks3050 

This looks like a missing test for the blobstore type as part of the conditional, fixing.

**Comment by bdellegrazie** [comment]
Hello @Aleks3050,

Your problem should be fixed in tag v3.24.0-02.2 or higher.
I have just pushed v3.25.0-03 too in case you want to use the latest Nexus.

Please reopen if you still experience an issue.
Thanks!

**Comment by Aleks3050** [comment]
ok! Thank you, I will test last version.

**Comment by Aleks3050** [comment]
Hi @bdellegrazie !
I try to use casc plugin for tag v3.24.0-02.2 
but see following error yet:
```
2020-07-24 14:08:57,677+0000 ERROR [FelixStartLevel] *SYSTEM com.weareadaptive.nexus.casc.plugin.internal.NexusCascPlugin - Failed transition: NEW -> STARTED
java.lang.NullPointerException: null
        at com.weareadaptive.nexus.casc.plugin.internal.NexusCascPlugin.lambda$4(NexusCascPlugin.java:236)
        at java.util.ArrayList.forEach(ArrayList.java:1257)
        at com.weareadaptive.nexus.casc.plugin.internal.NexusCascPlugin.applyRepositoryConfig(NexusCascPlugin.java:222)
        at com.weareadaptive.nexus.casc.plugin.internal.NexusCascPlugin.doStart(NexusCascPlugin.java:121)
        at org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport.start(StateGuardLifecycleSupport.java:69)
        at org.sonatype.nexus.common.stateguard.MethodInvocationAction.run(MethodInvocationAction.java:39)
        at org.sonatype.nexus.common.stateguard.StateGuard$TransitionImpl.run(StateGuard.java:193)
        at org.sonatype.nexus.common.stateguard.TransitionsInterceptor.invoke(TransitionsInterceptor.java:56)
        at org.sonatype.nexus.extender.NexusLifecycleManager.startComponent(NexusLifecycleManager.java:199)
        at org.sonatype.nexus.extender.NexusLifecycleManager.to(NexusLifecycleManager.java:111)
        at org.sonatype.nexus.extender.NexusContextListener.moveToPhase(NexusContextListener.java:321)
        at org.sonatype.nexus.extender.NexusContextListener.frameworkEvent(NexusContextListener.java:244)
        at org.apache.felix.framework.Felix.setActiveStartLevel(Felix.java:1431)
        at org.apache.felix.framework.FrameworkStartLevelImpl.run(FrameworkStartLevelImpl.java:308)
        at java.lang.Thread.run(Thread.java:748)
2020-07-24 14:08:57,679+0000 WARN  [FelixStartLevel] *SYSTEM org.sonatype.nexus.extender.NexusLifecycleManager - Problem starting TASKS: com.weareadaptive.nexus.casc.plugin.internal.NexusCascPlugin$$EnhancerByGuice$$5fcbb89e@7d618fd7
```

**Comment by bdellegrazie** [comment]
Hi @Aleks3050 ,

Apologies, I should have wrapped the entire statement in the if conditional rather than relying upon boolean short-circuit.
This is now fixed, please try either tag v3.24.0-02.3 or v3.25.0-03.1
